core-bundle/src/Resources/contao/modules/ModuleSearch.php

```diff
@@ -321,6 +325,48 @@ protected function compile()
 					$objTemplate->hasContext = true;
 				}
 
+				// Add image to search results
+				$objTemplate->hasImage = false;
+				if (isset($arrResult[$i]['meta']) && null !== $arrResult[$i]['meta'])
+				{
+					$meta = json_decode($arrResult[$i]['meta'], true);
+					// Reverse so the last element wins
+					$meta = array_reverse($meta);
+					foreach ($meta as $v)
+					{
+						if (!isset($v['https://schema.org/primaryImageOfPage']))
+						{
+							continue;
+						}
+
+						if ($figureBuilder = $this->getFigureBuilderIfResourceExists($v['https://schema.org/primaryImageOfPage']['contentUrl']))
+						{
+							$objTemplate->hasImage = true;
+
+							$figureMeta = array();
+							if (isset($v['https://schema.org/primaryImageOfPage']['caption']))
+							{
+								$figureMeta[Metadata::VALUE_CAPTION]  = $v['https://schema.org/primaryImageOfPage']['caption'];
+							}
+							if (isset($v['https://schema.org/primaryImageOfPage']['name']))
+							{
+								$figureMeta[Metadata::VALUE_TITLE]  = $v['https://schema.org/primaryImageOfPage']['name'];
+							}
+
+							if (isset($v['https://schema.org/primaryImageOfPage']['alternateName']))
+							{
+								$figureMeta[Metadata::VALUE_ALT]  = $v['https://schema.org/primaryImageOfPage']['alternateName'];
+							}
+
+							$objTemplate->image = (object) $figureBuilder
+								->setSize($this->imgSize)
+								->setMetadata(new Metadata($figureMeta))
+								->build()
+								->getLegacyTemplateData();
```

Comment:
Why not use `->applyLegacyTemplateData($objTemplate)` here?

Reply:
I wanted to make sure it's not top level but resides inside `->image` instead so that if others want to add their own images, it feels more consistent.